transport/src/main/java/io/netty/channel/nio/SelectedSelectionKeySet.java

```diff
@@ -107,4 +62,25 @@ public boolean contains(Object o) {
     public Iterator<SelectionKey> iterator() {
         throw new UnsupportedOperationException();
     }
+
+    void reset() {
+        reset(0);
+    }
+
+    void reset(int start) {
+        for (; start < size; ++start) {
+            keys[start] = null;
+        }
+        size = 0;
```

Comment:
no issue comes to mind at the moment, just unexpected behaviour that size of the set could be 0 when there are non null entries in the null.